packages/mui-base/test/createRenderer.ts

```diff
@@ -2,66 +2,26 @@ import {
   CreateRendererOptions,
   RenderOptions,
   createRenderer as sharedCreateRenderer,
-  act,
+  flushMicrotasks,
+  Renderer,
   MuiRenderResult,
+  act,
 } from '@mui/internal-test-utils';
 
-function safeAct(callback: () => void) {
-  if (navigator.userAgent.includes('jsdom')) {
-    return act(callback);
-  }
-
-  return callback();
-}
-
-interface Clock {
-  /**
-   * Runs all timers until there are no more remaining.
-   * WARNING: This may cause an infinite loop if a timeout constantly schedules another timeout.
-   * Prefer to to run only pending timers with `runToLast` and unmount your component directly.
-   */
-  runAll(): void;
-  /**
-   * Runs only the currently pending timers.
-   */
-  runToLast(): void;
-  /**
-   * Tick the clock ahead `timeoutMS` milliseconds.
-   * @param timeoutMS
-   */
-  tick(timeoutMS: number): void;
-  /**
-   * Returns true if we're running with "real" i.e. native timers.
-   */
-  isReal(): boolean;
-  /**
-   * Runs the current test suite (i.e. `describe` block) with fake timers.
-   */
-  withFakeTimers(): void;
-  /**
-   * Restore the real timer
-   */
-  restore(): void;
-}
-
-// TODO: replace with Renderer from @mui/internal-test-utils once it's exported
-interface CreateRendererResult {
+type BaseUITestRenderer = Omit<Renderer, 'render'> & {
   render: (element: React.ReactElement, options?: RenderOptions) => Promise<MuiRenderResult>;
-  clock: Clock;
-}
+};
 
-export function createRenderer(globalOptions?: CreateRendererOptions): CreateRendererResult {
+export function createRenderer(globalOptions?: CreateRendererOptions): BaseUITestRenderer {
   const createRendererResult = sharedCreateRenderer(globalOptions);
   const { render: originalRender } = createRendererResult;
 
-  const render = async (element: React.ReactElement, options?: RenderOptions) => {
-    const result = await originalRender(element, options);
-
-    // flush microtasks
-    await safeAct(() => async () => {});
-
-    return result;
-  };
+  const render = async (element: React.ReactElement, options?: RenderOptions) =>
+    act(async () => {
+      const result = await originalRender(element, options);
+      await flushMicrotasks();
```

Comment:
This looks strange. Why does Base UI needs this when the other repositories don't?

Reply:
Most of our components rely on Floating UI, which requires this pattern for testing in some cases. I figured it's better to always wait for pending promises instead of deciding on test-by-test basis (which also makes the test more close to what users perceive).